include/nbl/core/containers/LRUCache.h

```diff
@@ -16,206 +18,399 @@ namespace core
+				m_shortcut_map.erase(base_t::m_list.getLastAddress());
+				base_t::m_list.popBack();
 			}
+			if constexpr (std::is_same_v<Value, V>)
+				base_t::m_list.emplaceFront(std::forward<K>(k), std::forward<V>(v));
 			else
-				return nullptr;
+				base_t::m_list.emplaceFront(std::forward<K>(k), Value(std::forward<V>(v)));
+			m_shortcut_map.insert(base_t::m_list.getFirstAddress());
 		}
+		return &base_t::m_list.getBegin()->data.second;
+	}
 
-		//get the value from cache at an associated Key, or nullptr if Key is not contained within cache. Does not alter the value use order
-		inline const Value* peek(const Key& key) const
-		{
-			uint32_t i = common_peek(key);
-			if (i!=invalid_iterator)
-				return &(base_t::m_list.get(i)->data.second);
-			else
-				return nullptr;
-		}
-		inline Value* peek(const Key& key)
+	template<typename K, typename V>
+	inline Value* insert(K&& k, V&& v)
+	{
+		return insert(std::forward<K>(k), std::forward<V>(v), [](const Value& ejected)->void {});
+	}
+
+	//get the value from cache at an associated Key, or nullptr if Key is not contained within cache. Marks the returned value as most recently used
+	inline Value* get(const Key& key)
+	{
+		auto i = common_peek(key);
+		if (i != invalid_iterator)
 		{
-			uint32_t i = common_peek(key);
-			if (i != invalid_iterator)
-				return &(base_t::m_list.get(i)->data.second);
-			else
-				return nullptr;
+			base_t::m_list.moveToFront(i);
+			return &(base_t::m_list.get(i)->data.second);
 		}
+		else
+			return nullptr;
+	}
 
-		//remove element at key if present
-		inline void erase(const Key& key)
+	//get the value from cache at an associated Key, or nullptr if Key is not contained within cache. Does not alter the value use order
+	inline const Value* peek(const Key& key) const
+	{
+		uint32_t i = common_peek(key);
+		if (i != invalid_iterator)
+			return &(base_t::m_list.get(i)->data.second);
+		else
+			return nullptr;
+	}
+	inline Value* peek(const Key& key)
+	{
+		uint32_t i = common_peek(key);
+		if (i != invalid_iterator)
+			return &(base_t::m_list.get(i)->data.second);
+		else
+			return nullptr;
+	}
+
+	//remove element at key if present
+	inline void erase(const Key& key)
+	{
+		bool success;
+		shortcut_iterator_t iterator = common_find(key, success);
+		if (success)
 		{
-			bool success;
-			shortcut_iterator_t iterator = common_find(key,success);
-			if (success)
-			{
-				base_t::m_list.erase(*iterator);
-				m_shortcut_map.erase(iterator);
-			}
+			base_t::m_list.erase(*iterator);
+			m_shortcut_map.erase(iterator);
 		}
+	}
 
-	protected:
-		unordered_set<uint32_t,WrapHash,WrapEquals> m_shortcut_map;
-};
+	/**
+	* @brief Resizes the cache by extending its capacity so it can hold more elements. Returns a bool indicating if capacity was indeed increased.
+	*
+	* @param [in] newCapacity New number of elements to hold. MUST be greater than current cache capacity.
+	*/
+	inline bool grow(uint32_t newCapacity)
+	{
+		m_shortcut_map.reserve(newCapacity);
```

Comment:
let's check for newCapacity being more than previous one here as well.
we don't know what these containers will be or how they'll behave in the future